google/cloud/testing_util/BUILD.bazel

```diff
@@ -12,7 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-package(default_visibility = ["//visibility:public"])
+# TODO(#3701) Change the default_visibility to "//visibility:private", and
+# widen visibility as needed with per-target visibility.
+package(
+    default_deprecation = """
+    These targets are not for public use and will be removed sometime around
+    July 1, 2022. Only use Bazel targets defined in the top-level //BUILD.bazel
+    file. See https://github.com/googleapis/google-cloud-cpp/issues/3701 for
+    more details.
+    """,
+    default_visibility = ["//visibility:public"],
+)
```

Comment:
Argh, also here, there are rules here that are already deprecated.

Reply:
Yes, I saw those. But to keep things simpler, I was leaning toward just handling these deprecations as whole files, even if an already deprecated thing stays around a bit longer. I could remove the other deprecation notices though.

Reply:
Delaying the deprecation to simplify things SGTM.  Removing them right away also SGTM.